slack_sdk/models/blocks/block_elements.py

```diff
@@ -1381,6 +1405,185 @@ def __init__(
+        return super().attributes.union(
+            {
+                "initial_value",
+                "dispatch_action_config",
+            }
+        )
+
+    def __init__(
+        self,
+        *,
+        action_id: Optional[str] = None,
+        initial_value: Optional[str] = None,
+        dispatch_action_config: Optional[Union[dict, DispatchActionConfig]] = None,
+        focus_on_load: Optional[bool] = None,
+        placeholder: Optional[Union[str, dict, TextObject]] = None,
+        **others: dict,
+    ):
+        """
+        A URL input element, similar to the Plain-text input element,
+        creates a single line field where a user can enter URL-encoded data.
+        https://api.slack.com/reference/block-kit/block-elements#url
+
+        Args:
+            action_id (required): An identifier for the input value when the parent modal is submitted.
+                You can use this when you receive a view_submission payload to identify the value of the input element.
+                Should be unique among all other action_ids in the containing block.
+                Maximum length for this field is 255 characters.
+            initial_value: The initial value in the URL input when it is loaded.
+            dispatch_action_config: A dispatch configuration object that determines when during text input
+                the element returns a block_actions payload.
+            focus_on_load: Indicates whether the element will be set to auto focus within the view object.
+                Only one element can be set to true. Defaults to false.
+            placeholder: A plain_text only text object that defines the placeholder text shown in the URL input.
+                Maximum length for the text in this field is 150 characters.
+        """
+        super().__init__(
+            type=self.type,
+            action_id=action_id,
+            placeholder=TextObject.parse(placeholder, PlainTextObject.type),
+            focus_on_load=focus_on_load,
+        )
+        show_unknown_key_warning(self, others)
+
+        self.initial_value = initial_value
+        self.dispatch_action_config = dispatch_action_config
+
+
+# -------------------------------------------------
+# Number Input Element
+# -------------------------------------------------
+
+
+class NumberInputElement(InputInteractiveElement):
+    type = "number_input"
+
+    @property
+    def attributes(self) -> Set[str]:
+        return super().attributes.union(
+            {
+                "initial_value",
+                "is_decimal_allowed",
+                "min_value",
+                "max_value",
+                "dispatch_action_config",
+            }
+        )
+
+    def __init__(
+        self,
+        *,
+        action_id: Optional[str] = None,
+        is_decimal_allowed: Optional[bool] = None,
+        initial_value: Optional[str] = None,
+        min_value: Optional[str] = None,
+        max_value: Optional[str] = None,
```

Comment:
nice-to-have:
We can accept these three types and call `str()` when assigning to instance fields.
```suggestion
        initial_value: Optional[Union[str, int, float]] = None,
        min_value: Optional[Union[str, int, float]] = None,
        max_value: Optional[Union[str, int, float]] = None,
```

Reply:
This is interesting to me as it's the same suggestion I had in the Java PR: https://github.com/slackapi/java-slack-sdk/pull/1076#discussion_r1008171285

So should we provide this kind of 'helpful' interface to developers in all our SDKs? The fact that underlying HTTP API uses strings is more coincidence and a backend implementation/runtime constraint than actually a consideration for external developers.

For what it is worth IMO I think it is a nice addition, though it does increase the complexity a little bit in the SDK side.

Reply:
We could leave the type as `Optional[str]` since this is what is [in the docs](https://api.slack.com/reference/block-kit/block-elements#number), but still call `str()` when assigning to instance fields.

This way our types follow the docs and a dev that passes an `int` or `float` won't get a runtime error from the slack api.

what do you think @filmaj @seratch?

Reply:
That sounds good to me and was what I hinted at yesterday in our discussion with the backend team as to what we could do on the SDK side, but I would want to make sure @seratch (or, for that matter, anyone else on our team) has no issues with that approach.

Reply:
My suggestion here is that, regardless of the types given by a developer, this SDK eventually sends str values to the server-side. When it comes to the use cases where a developer directly manipulates dict objects, we should not (and actually are unable to) support int/float types for sure. However, these classes are supposed to be provide better developer experience. So, enabling developers to easily use more appropriate types should make sense.